### ISSUE TITLE
Solution based restore does not restore the projects with solution type dependencies

### ISSUE THREAD

**Issue by cdmihai** [NuGet/Home#5610]
Please read the following information before posting the issue.

## Details about Problem

NuGet product used (dotnet.exe | `msbuild /t:restore`):

dotnet.exe --version (if appropriate): 1.0.4

msbuild /version: 15.3.402.46929

VS version (if appropriate): 15.0.26709.3000

OS version (i.e. win10 v1607 (14393.321)): Version	10.0.15063 Build 15063

## Detailed repro steps so we can see the same problem

1. Create a solution with two .net core projects (A and B). 

2. Create a solution type dependency between A and B: right click on solution -> "Project Dependencies .." -> make A depend on B
![image](https://user-images.githubusercontent.com/2255729/28291628-1e41f236-6b00-11e7-8d7f-4ac42deb8e9a.png)

This makes the solution file look something like this:
```
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "A", "A\A.csproj", "{F82B0350-2791-4CF4-A860-1D18876F5B5B}"
	ProjectSection(ProjectDependencies) = postProject
		{FDC88EB1-A799-45EB-86CC-ACA649F31E4B} = {FDC88EB1-A799-45EB-86CC-ACA649F31E4B}
	EndProjectSection
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "B", "B\B.csproj", "{FDC88EB1-A799-45EB-86CC-ACA649F31E4B}"
EndProject
```

3. Both A and B restore under VS

4. In a VS command prompt: `msbuild /t:restore SolutionWithProjectDependencies.sln`.  Both A and B restore

5. In a VS command prompt: `dotnet restore SolutionWithProjectDependencies.sln`. Only B restores.

Expected:
Both A and B restore after `dotnet restore SolutionWithProjectDependencies.sln`.

This causes `dotnet build` to fail because project A is not restored.
...

### Sample Project
[SolutionWithProjectDependencies.zip](https://github.com/NuGet/Home/files/1154136/SolutionWithProjectDependencies.zip)

**Comment by cdmihai** [comment]
Found this out while looking into https://github.com/Microsoft/msbuild/issues/2274. This nuget issue does not appear to be connected to it, since `msbuild /t:restore` works OK.

**Comment by cdmihai** [comment]
Appears to work on dotnet 2.0.0-preview3-006770. Feel free to close the issue.

**Comment by emgarten** [comment]
> Appears to work on dotnet 2.0.0-preview3-006770. Feel free to close the issue.

Thanks for checking, this was fixed in NuGet 4.3.0 which is in 2.0.0 and 1.1.0.
